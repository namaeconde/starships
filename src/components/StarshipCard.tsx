import React, { useState } from "react";
import Box from "@mui/material/Box";
import Typography from "@mui/material/Typography";
import Rating from '@mui/material/Rating';
import {color} from '../theme';
import IconButton from "@mui/material/IconButton";
import FavoriteIcon from '@mui/icons-material/Favorite';
import FavoriteBorderIcon from '@mui/icons-material/FavoriteBorder';
import StarshipImage from '../images/starship.png';
import Grid from "@mui/material/Grid";
import TextField from '@mui/material/TextField';
import { useDispatch } from 'react-redux';
import { addToList, removeFromList, updateFavorite } from '../redux/favoriteListSlice';
import LightTooltip from "./LightTooltip";

export default function StarshipCard({ starship, favorite, showNote }: { starship: any, favorite: boolean, showNote: boolean }): JSX.Element { // TODO: Update starship any to Starship Interface
  const { name, manufacturer, hyperdrive_rating, passengers } = starship;
  const [ isFavorite, setIsFavorite ] = useState(favorite);
  const [ notes, setNotes ] = useState(starship.notes);
  const dispatch = useDispatch();

  const onSetFavorite = () => {
    if (isFavorite) {
      dispatch(removeFromList(starship))
    } else {
      dispatch(addToList(starship));
    }
    setIsFavorite(!isFavorite);
  }

  const handleAddNotes = (event: { target: { value: any; }; }) => {
    const newNote = event.target.value;
    setNotes(newNote);
    dispatch(updateFavorite({...starship, notes: newNote }));
  }

  return (
    <Box sx={{ p:3, backgroundColor:`${color.midnigt_moss}`, borderRadius:2 }}>
      <Box container component={Grid}>
        <Box item xs={8} component={Grid}>
          <LightTooltip title={name}>
            <Typography noWrap fontSize={24} fontWeight={700}>{name}</Typography>
          </LightTooltip>
          <LightTooltip title={manufacturer}>
          <Typography noWrap fontSize={18}>{manufacturer}</Typography>
          </LightTooltip>
          <Rating name="read-only" value={Number(hyperdrive_rating)} precision={0.5} readOnly />
          <Typography fontSize={18}>Passengers: {passengers}</Typography>
        </Box>
        <Box item xs={4} sx={{ backgroundImage: `url(${StarshipImage})`, borderRadius:2 }} component={Grid}>
          <IconButton color="primary" sx={{ float:"right" }} onClick={onSetFavorite}>
            { isFavorite ? <FavoriteIcon/> : <FavoriteBorderIcon/>}
          </IconButton>
        </Box>
      </Box>
      {
        showNote && 
        <Box 
          sx={{
            width: 500,
            maxWidth: '100%',
          }} 
          mt={2}
        >
          <TextField 
            fullWidth
            id="notes" 
            label="Add text" 
            variant="outlined" 
            multiline
            rows={4}
            value={notes}
            onChange={handleAddNotes}
          />
        </Box>
      }
    </Box>
  )
}