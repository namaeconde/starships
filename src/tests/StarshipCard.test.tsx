import React from 'react';
import { render, cleanup } from '@testing-library/react'
import StarshipCard, { FavoriteStarshipCard } from '../components/StarshipCard';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

let starship = {
  name: "CR90 corvette",
  manufacturer: "Corellian Engineering Corporation",
  hyperdrive_rating: 2.0,
  passengers: "600",
  notes: "I love the design on this one! ðŸš€"
}
const initialState = { favoriteList: { value: [ starship ] } }
const mockStore = configureStore()
let store

afterEach(cleanup);

test('StarshipCard renders', () => {
  store = mockStore(initialState);
  const { asFragment } = render(<Provider store={store}><StarshipCard starship={starship}/></Provider>);
  expect(asFragment()).toMatchSnapshot();
});

test('FavoriteStarshipCard renders', () => {
  store = mockStore(initialState);
  const { asFragment } = render(<Provider store={store}><FavoriteStarshipCard starship={starship}/></Provider>);
  expect(asFragment()).toMatchSnapshot();
});

