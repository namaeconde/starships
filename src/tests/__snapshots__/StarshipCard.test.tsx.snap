// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteStarshipCard renders 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-5suu7y"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined MuiBox-root css-1uvpa41-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiBox-root css-1h3zkf6-MuiGrid-root"
      >
        <p
          aria-label="CR90 corvette"
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-tt8b08-MuiTypography-root"
          data-mui-internal-clone-element="true"
        >
          CR90 corvette
        </p>
        <p
          aria-label="Corellian Engineering Corporation"
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-f4vo41-MuiTypography-root"
          data-mui-internal-clone-element="true"
        >
          Corellian Engineering Corporation
        </p>
        <span
          aria-label="2 Stars"
          class="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly css-ryrseu-MuiRating-root"
          role="img"
        >
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
        >
          Passengers: 600
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiBox-root css-1ygt30y-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-t7j1t6-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FavoriteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root css-1xossyd"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="notes"
          id="notes-label"
        >
          Add text
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-8ewcdo-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <textarea
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
            id="notes"
            rows="4"
            style="height: 0px; overflow: hidden;"
          >
            I love the design on this one! 🚀
          </textarea>
          <textarea
            aria-hidden="true"
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline css-1sqnrkk-MuiInputBase-input-MuiOutlinedInput-input"
            readonly=""
            style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
            tabindex="-1"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-186xcr5"
            >
              <span>
                Add text
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`StarshipCard renders 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-5suu7y"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined MuiBox-root css-1uvpa41-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiBox-root css-1h3zkf6-MuiGrid-root"
      >
        <p
          aria-label="CR90 corvette"
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-tt8b08-MuiTypography-root"
          data-mui-internal-clone-element="true"
        >
          CR90 corvette
        </p>
        <p
          aria-label="Corellian Engineering Corporation"
          class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-f4vo41-MuiTypography-root"
          data-mui-internal-clone-element="true"
        >
          Corellian Engineering Corporation
        </p>
        <span
          aria-label="2 Stars"
          class="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly css-ryrseu-MuiRating-root"
          role="img"
        >
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
          </span>
          <span
            class="MuiRating-decimal css-34he1w-MuiRating-decimal"
          >
            <span
              style="width: 0%; overflow: hidden; z-index: 1; position: absolute;"
            >
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
            <span>
              <span
                class="MuiRating-icon MuiRating-iconEmpty css-1c99szj-MuiRating-icon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                  data-testid="StarBorderIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
                  />
                </svg>
              </span>
            </span>
          </span>
        </span>
        <p
          class="MuiTypography-root MuiTypography-body1 css-qscxtw-MuiTypography-root"
        >
          Passengers: 600
        </p>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 MuiBox-root css-1ygt30y-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-t7j1t6-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="FavoriteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
